src/main/java/duke/Lister.java

```diff
@@ -0,0 +1,117 @@
+package duke;
+
+/**
+ * The Lister Class parses commands and references TaskList and Storage Classes to store and retrieve Tasks
+ * @author qiujingying
+ * @version 1.0
+ */
+public class Lister {
+    private TaskList taskList;
+    private Storage storage;
+
+    /**
+     * Creates a Lister object to reference to TaskList and Storage objects
+     * @param tasks initial TaskList stores Tasks
+     * @param storage initial storage with filepath to store data
+     */
+    public Lister(TaskList tasks, Storage storage) {
+        this.taskList = tasks;
+        this.storage = storage;
+    }
+
+    /**
+     * Parses commands from command line
+     * @param command input to identify relevant Task objects to create and their description
+     */
+    public void record(String command) {
+        if (command.equals("list")) {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < taskList.getSize(); i++) {
+                System.out.println((i + 1) + "." + taskList.retrieveTask(i).toString());
+            }
+        } else {
+            try {
+                int x = getIndex(command);
```

Comment:
Try not to use vague variables like x anywhere in your code, even if it's meant for indexing. Using something like index or idx will be clearer :))